### ISSUE TITLE
Run the script when Lan connection is detected?

### ISSUE THREAD

**Issue by AlirezaXdehghan** [FalsePhilosopher/PPPwnWRT#22]
Hi there, thanks for the great writeup.
I was thinking is there a way to only run the script when a connection is detected on the interface (e.g br-lan) instead of running it every X seconds?
because the thing is I want to make it fully automated but it keeps running in the background even after pwning the ps4 so it leads to crashes because it tries to corrupt an object in the memory.
or is there a plugin to stop network activity in ps4 when goldhen is launched?
my goal is to make this fully automated without doing anything on user's end!
thanks.

**Comment by FalsePhilosopher** [comment]
I think such functionality exists in the pi-pppwn setup and can be adapted to work for openwrt, i'm not sure the loop in the run script is necessary with the updates to the functionality of pppwn_cpp anyway. If you get an updated launch script working feel free to send a PR and add yourself to the contributor list.

**Comment by AlirezaXdehghan** [comment]
well thanks for pointing me to pi-pwn, actually their implementation of the detection was quite logical and easy! but they're using nmap to detect goldhen so I don't think it's practical to run it on a router with low flash storage, anyway I might start writing some code, thanks!

**Comment by FalsePhilosopher** [comment]
Anytime! I look forward to seeing what solutions you can throw at it!

**Comment by kadom11** [comment]
Hello my friend, I use a Xiaomi 4c router, and when I use the mips build, it does not work well, and sometimes it works. If possible, explain the method to me better, and does the type of openwrt affect it if the version is updated? Thanks.

**Comment by FalsePhilosopher** [comment]
> Hello my friend, I use a Xiaomi 4c router, and when I use the mips build, it does not work well, and sometimes it works. If possible, explain the method to me better, and does the type of openwrt affect it if the version is updated? Thanks.

Not an issue with PPPwnWRT but with pppwn_cpp, open an issue in pppwn_cpp if you are having troubles with using pppwn_cpp.

**Comment by kadom11** [comment]
thanks for your reply . But when I load the mips bulid, it appears like this and in red 

change stage1.bin, firmware version and interface accordingly in '/etc/pppwnwrt/pppwnwrt.sh'! start the script after you change these variables!

**Comment by FalsePhilosopher** [comment]
> thanks for your reply . But when I load the mips bulid, it appears like this and in red
> 
> change stage1.bin, firmware version and interface accordingly in '/etc/pppwnwrt/pppwnwrt.sh'! start the script after you change these variables!

That's correct, it is part of the install process of step 4 notifying you of what interfaces you have available for you to use in step 5 and is not an error. If you need further assistance on installing or troubleshooting you can ask in the playstation homebrew discord ps4 channel and they are very fast to respond.

**Comment by FalsePhilosopher** [comment]
> when a connection is detected on the interface (e.g br-lan)

Asking chatgpt about pppoe info it spit out

For detecting a PPPoE PADI request and checking if port 3232 is open without using `tcpdump`, `nmap`, or other external tools, you'll need to rely on native tools available in OpenWRT while keeping the script lightweight.

### Suggested Approach:

1. **Monitoring for PADI Requests**:
   - Instead of `tcpdump`, OpenWRT has a native `logread` command, which reads system logs. Since PPPoE PADI requests are part of the network activity, OpenWRT’s logs should capture them. You can monitor logs for relevant PPPoE events.
   
2. **Port Scan**:
   - For checking if port 3232 is open without `nmap`, you can use the built-in `netcat` (`nc`) which is available on OpenWRT by default and is very lightweight.

### Updated Script:

```bash
#!/bin/bash

# Define variables
PORT=3232
PAYLOAD="your_payload.sh"  # Replace with your payload script
PPP_IFACE="ppp0"           # PPPoE interface; adjust as needed
LOG_KEYWORD="PADI"         # Log keyword to detect PADI request

# Function to send the payload
send_payload() {
    echo "Port $PORT is closed. Sending payload..."
    bash "$PAYLOAD"
}

# Function to check if port 3232 is open
check_port() {
    echo "Checking if port $PORT is open on $PPP_IFACE..."
    nc -z -w 2 localhost "$PORT" &>/dev/null
    if [ $? -eq 0 ]; then
        echo "Port $PORT is open. Skipping payload..."
        exit 0
    else
        send_payload
    fi
}

# Function to monitor for PPPoE PADI request in system logs
monitor_padi_request() {
    echo "Waiting for PPPoE PADI request on $PPP_IFACE..."
    while true; do
        logread | grep -q "$LOG_KEYWORD"
        if [ $? -eq 0 ]; then
            echo "PADI request detected. Performing port check..."
            check_port
            break
        fi
        sleep 1
    done
}

# Main logic
echo "Waiting for PPPoE connection on $PPP_IFACE..."
while ! ip a | grep -q "$PPP_IFACE"; do
    sleep 1
done

# Start monitoring for PADI requests
monitor_padi_request
```

### Key Details:
1. **PADI Request Detection**: The script uses `logread`, a native OpenWRT tool, to monitor system logs for PPPoE PADI-related events. The specific keyword used (`PADI`) can be adjusted based on the actual log entries that OpenWRT generates when PPPoE connections are initiated.
   
2. **Port Check**: It uses `nc` (Netcat) to check whether port 3232 is open on the local interface. If the port is closed, the script sends the payload.

3. **Minimal Storage Impact**: This approach avoids large tools like `tcpdump` or `nmap`, making it suitable for routers with limited storage.

You can adjust the `LOG_KEYWORD` to match specific log entries from OpenWRT related to PPPoE PADI requests, depending on the exact logging behavior of your router. This should work in a more constrained OpenWRT environment.

**Comment by FalsePhilosopher** [comment]
I compiled the most stripped down version of nmap with `./configure --without-subversion --without-liblua --without-zenmap --with-pcre=/usr --with-libpcap=included --with-libdnet=included --without-ndiff --without-nmap-update --without-ncat --without-liblua --without-nping --without-openssl` then using `upx --best` compressed the executable down to 5mb. Along with the required supporting libs that might still be too much for low end, it might just have to be a separate section of the readme for cross compilation/compression and enabling such feature.

**Comment by FalsePhilosopher** [comment]
This is now fixed with the latest goldhen disabling additional attempts to root the console once it's rooted, so make sure you use the latest GH bin to avoid this issue.
